symplyphysics/laws/thermodynamics/internal_energy_differential.py

```diff
@@ -19,34 +20,41 @@
 ## The fundamental thermodynamic relation are fundamental equations which demonstate how important
 ## thermodynamic quantities depend on variables that are measurable experimentally.
 
-# Law: dU = T * dS - P * dV
+# Law: dU = T * dS - P * dV + mu * dN
 ## U - internal energy
 ## T - absolute temperature
 ## S - entropy
 ## P - pressure
 ## V - volume
+## mu - chemical potential
+## N - number of particles in the system
 ## Notation: d(x) - full differential of `x`
 
 # Note
-## - Entropy `S` and volume `V` are so called natural variables of internal energy as a
+## - Entropy `S`, volume `V` and particle count `N` are so called natural variables of internal energy as a
 ##   thermodynamic potential.
+## - For a system with more than one type of particles, the last term can be represented as a sum over all
+##   types of particles, i.e. `Sum(mu_i * d(N_i), i)`.
 
 # Conditions
-## - The system is closed
 ## - The system is in thermal equilibrium with its surroundings
+## - There is only one type of particles in the system.
 
 internal_energy_change = Symbol("internal_energy_change", units.energy)
 temperature = symbols.thermodynamics.temperature
 entropy_change = Symbol("entropy_change", units.energy / units.temperature)
 pressure = Symbol("pressure", units.pressure)
 volume_change = Symbol("volume_change", units.volume)
+chemical_potential = Symbol("chemical_potential", units.energy)
+particle_count_change = Symbol("particle_count_change", dimensionless)
 
 law = Eq(
     internal_energy_change,
-    temperature * entropy_change - pressure * volume_change,
+    temperature * entropy_change - pressure * volume_change + chemical_potential * particle_count_change,
 )
 
 # Derive from the first and second laws of thermodynamics
+# Assuming the particle count does not change
```

Comment:
You are deriving a law for the closed system. You should either update proof for general form of the law; or add TODO and fix later; or revert refactor; or remove proof.

Reply:
Okay, I'll add a TODO